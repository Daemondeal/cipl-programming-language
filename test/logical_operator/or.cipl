print(1 or true)
print(false or 1)
print(false or false or true)

print(false or false)
print(false or false or false)
print(false or false or null)

let a = "before"
let b = "before"
(a = false) or (b = true) or (a = "bad")

print(a)
print(b)
