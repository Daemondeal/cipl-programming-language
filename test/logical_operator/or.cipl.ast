(expression (call print (or 1 true)))
(expression (call print (or false 1)))
(expression (call print (or (or false false) true)))
(expression (call print (or false false)))
(expression (call print (or (or false false) false)))
(expression (call print (or (or false false) null)))
(let a before)
(let b before)
(expression (or (or (group (assign a false)) (group (assign b true))) (group (assign a bad))))
(expression (call print a))
(expression (call print b))

