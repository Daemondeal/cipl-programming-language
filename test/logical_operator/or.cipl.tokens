Token          Lexeme         Line 
IDENTIFIER     print          1    
LEFT_PAREN     (              1    
NUMBER         1              1     1
OR             or             1    
TRUE           true           1    
RIGHT_PAREN    )              1    
LINE_END       \n             2    
IDENTIFIER     print          2    
LEFT_PAREN     (              2    
FALSE          false          2    
OR             or             2    
NUMBER         1              2     1
RIGHT_PAREN    )              2    
LINE_END       \n             3    
IDENTIFIER     print          3    
LEFT_PAREN     (              3    
FALSE          false          3    
OR             or             3    
FALSE          false          3    
OR             or             3    
TRUE           true           3    
RIGHT_PAREN    )              3    
LINE_END       \n             4    
IDENTIFIER     print          5    
LEFT_PAREN     (              5    
FALSE          false          5    
OR             or             5    
FALSE          false          5    
RIGHT_PAREN    )              5    
LINE_END       \n             6    
IDENTIFIER     print          6    
LEFT_PAREN     (              6    
FALSE          false          6    
OR             or             6    
FALSE          false          6    
OR             or             6    
FALSE          false          6    
RIGHT_PAREN    )              6    
LINE_END       \n             7    
IDENTIFIER     print          7    
LEFT_PAREN     (              7    
FALSE          false          7    
OR             or             7    
FALSE          false          7    
OR             or             7    
NULL           null           7    
RIGHT_PAREN    )              7    
LINE_END       \n             8    
LET            let            9    
IDENTIFIER     a              9    
EQUAL          =              9    
STRING         "before"       9     before
LINE_END       \n             10   
LET            let            10   
IDENTIFIER     b              10   
EQUAL          =              10   
STRING         "before"       10    before
LINE_END       \n             11   
LEFT_PAREN     (              11   
IDENTIFIER     a              11   
EQUAL          =              11   
FALSE          false          11   
RIGHT_PAREN    )              11   
OR             or             11   
LEFT_PAREN     (              11   
IDENTIFIER     b              11   
EQUAL          =              11   
TRUE           true           11   
RIGHT_PAREN    )              11   
OR             or             11   
LEFT_PAREN     (              11   
IDENTIFIER     a              11   
EQUAL          =              11   
STRING         "bad"          11    bad
RIGHT_PAREN    )              11   
LINE_END       \n             12   
IDENTIFIER     print          13   
LEFT_PAREN     (              13   
IDENTIFIER     a              13   
RIGHT_PAREN    )              13   
LINE_END       \n             14   
IDENTIFIER     print          14   
LEFT_PAREN     (              14   
IDENTIFIER     b              14   
RIGHT_PAREN    )              14   
LINE_END       \n             15   
EOF                           15   
