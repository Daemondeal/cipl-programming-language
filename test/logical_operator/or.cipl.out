::stdout::
1
1
true
false
false
null
false
true

::stderr::
